searxng/searxng
# [fix] simple theme: make JS sources available on non build hosts

## What does this PR do?

[fix] simple theme: make JS sources available on non build hosts

## Why is this change important?

Include the source files directly in the `.map` files [1] for two reasons:

1. In the simple theme, modules from `./node_modules` are only available on
   developer's build host (a `make node.env` is needed to get the sources).  By
   example; in the `searxng.min.js.map` file, the `autocomplete.js` is linked at
   URL:

        ../node_modules/autocomplete-js/dist/autocomplete.js

2. Later, when client sources has been moved out of the `./static` URL [2] to
   the `<repo-roo>/src` folder, all JS sources are no longer available.

[1] https://github.com/gruntjs/grunt-contrib-uglify#sourcemapincludesources
[2] https://github.com/searxng/searxng/issues/141

## How to test this PR locally?

    make node.clean run

In the browser open debug mode and see if you have access to the JS source of
e.g. `autocomplete.js`

    /static/themes/simple/node_modules/autocomplete-js/dist/autocomplete.js

## Author's checklist

Drawback: `.map` files grow up in size, but this should not bother, because the
files are only loaded in debug mode.
